src/MSBuildBinLogQuery/Component/Project.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Logging.Query.Graph;
+using Microsoft.Build.Logging.Query.Utility;
+
+namespace Microsoft.Build.Logging.Query.Component
+{
+    public class Project : Component, IEquatable<Project>
+    {
+        public int Id { get; }
+        public string ProjectFile { get; }
+        public Build ParentBuild { get; }
+        public ItemManager Items { get; }
+        public PropertyManager Properties { get; }
+        public PropertyManager GlobalProperties { get; }
+        public ConcurrentDictionary<string, Target> TargetsByName { get; }
+        public ConcurrentDictionary<int, Target> TargetsById { get; }
+        // TODO: A single project instance can be built concurrently with
+        //       distinct EntryPointTargets. Those should be tracked as part of
+        //       the overall Project.
+        public IReadOnlyList<Target> EntryPointTargets { get; }
+        public ConcurrentBag<Target> OrderedTargets { get; }
+        public ProjectNode_BeforeThis Node_BeforeThis { get; }
+
+        public Project(
+            int id,
+            string projectFile,
+            string targetNames,
+            IEnumerable items,
+            IEnumerable properties,
+            IDictionary<string, string> globalProperties,
+            Build parentBuild) : base()
+        {
+            Id = id;
+            ProjectFile = projectFile;
+            ParentBuild = parentBuild;
+            Items = new ItemManager();
+            Properties = new PropertyManager();
+            GlobalProperties = new PropertyManager();
+            TargetsByName = new ConcurrentDictionary<string, Target>();
+            TargetsById = new ConcurrentDictionary<int, Target>();
+            EntryPointTargets = new List<Target>(
+                targetNames
+                .Split(';')
+                .Where(name => !string.IsNullOrWhiteSpace(name.Trim()))
+                .Select(name => GetOrAddTargetWithName(name.Trim())));
```

Comment:
This feels like a problem. Won't it mean that we don't get the real information on any target that was in the entry-point list?

I think the easiest thing to do for now is just keep this as a `List<string>`. We can consider making it a link to live objects later (that's a nicer API) but it's easy to correlate if you really need to.

Reply:
I actually debated on this. This doesn't mean that we don't get real information on entry-point targets, because their contents will be updated when `TargetStartedEventArgs` is encountered. However, I now change my mind that a string list might be better because then there will be no chance for a target to be meaningless temporarily... I'll change this to a string list.

Reply:
Done.